Source/Particles/Collision/BackgroundStopping/HybridResistiveDrag.cpp

```diff
@@ -0,0 +1,176 @@
+        m_type = HybridResistiveDragType::COLLISION_FREQUENCY;
+    } else {
+        WARPX_ABORT_WITH_MESSAGE("expression_type for hybrid_resistive_drag must be either resisistivity or collision_frequency.");
+    }
+
+    // Set up parser for eta with rho, v_i, v_e, T_i, T_e, B inputs
+    const std::string & function_string;
+    pp_collision_name.query("drag_function(rho,v_i,v_e,T_i,T_e,B)", function_string);
+    m_expression_parser = utils::parser::makeParser(function_string,{"rho","v_i","v_e","T_i","T_e","B"});
+    m_expression_func = m_expression_parser->compile<6>();
+}
+
+void
+HybridResistiveDrag::doCollisions (amrex::Real cur_time, amrex::Real dt, MultiParticleContainer* mypc)
+{
+    WARPX_PROFILE("HybridResistiveDrag::doCollisions()");
+    using namespace amrex::literals;
+
+    // Iterate over each species specified in the collision operator
+    for (auto & species_name : m_species_names) {
+        auto& species = mypc->GetParticleContainerFromName(species_name);
+        amrex::ParticleReal const species_mass = species.getMass();
+        amrex::ParticleReal const species_charge = species.getCharge();
+
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(species_mass > 0_prt, "Error: With hybrid resistive drag, the species mass must be > 0");
+        WARPX_ALWAYS_ASSERT_WITH_MESSAGE(species_charge != 0_prt, "Error: With hybrid resistive drag, the species charge must be nonzero");
+
+        // Loop over refinement levels
+        auto const flvl = species.finestLevel();
+        for (int lev = 0; lev <= flvl; ++lev) {
+
+            auto *cost = WarpX::getCosts(lev);
+
+            // loop over particles box by box
+#ifdef _OPENMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+            for (WarpXParIter pti(species, lev); pti.isValid(); ++pti) {
+                if (cost && WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Timers)
+                {
+                    amrex::Gpu::synchronize();
+                }
+                auto wt = static_cast<amrex::Real>(amrex::second());
+
+                doHybridResistiveDragOnIonsWithinTile(pti, dt, cur_time, species_mass, species_charge);
+
+                if (cost && WarpX::load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::Timers)
+                {
+                    amrex::Gpu::synchronize();
+                    wt = static_cast<amrex::Real>(amrex::second()) - wt;
+                    amrex::HostDevice::Atomic::Add(&(*cost)[pti.index()], wt);
+                }
+            }
+
+        }
+    }
+}
+
+void HybridResistiveDrag::doHybridResistiveDragOnIonsWithinTile (WarpXParIter& pti, amrex::Real dt, amrex::Real t,
+                                                               amrex::ParticleReal species_mass, amrex::ParticleReal species_charge)
+{
+    using namespace amrex::literals;
+
+    // So that CUDA code gets its intrinsic, not the host-only C++ library version
```

Comment:
Since WarpX runs on other GPU's beside nvidia:
```suggestion
    // So that GPU code gets its intrinsic, not the host-only C++ library version
```